sound/soc/sof/ipc3-topology.c

```diff
@@ -2167,8 +2178,19 @@ static int sof_ipc3_dai_config(struct snd_sof_dev *sdev, struct snd_sof_widget *
 	case SOF_DAI_INTEL_ALH:
 		if (data) {
 			/* save the dai_index during hw_params and reuse it for hw_free */
-			if (flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS)
-				config->dai_index = data->dai_index;
+			if (flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS) {
+				/*
+				 * DAI 2 is used as the first DAI for SDW in topology, normalize
+				 * it to 0
```

Comment:
Maybe refer to sof_sdw.c (something like "see SOC_SDW_INTEL_BIDIR_PDI_BASE in sof_sdw.c") defitinion to explain where the value "2" comes from?

Reply:
> Maybe refer to sof_sdw.c (something like "see SOC_SDW_INTEL_BIDIR_PDI_BASE in sof_sdw.c") defitinion to explain where the value "2" comes from?

Added. But on the second thought, this is Intel specific. Should we do it here?

Reply:
Added sdw_bidir_base in sof_dev_desc struct. But I am not sure if it is the best place to add the base.